tools/clang/lib/Parse/ParseDecl.cpp

```diff
@@ -397,9 +397,19 @@ bool Parser::MaybeParseHLSLAttributes(std::vector<hlsl::UnusualAnnotation *> &ta
           stage = hlsl::DXIL::PayloadAccessShaderStage::Miss;
         } else if (shaderStage == "anyhit") {
           stage = hlsl::DXIL::PayloadAccessShaderStage::Anyhit;
-        } 
+        }
+
+        // mod.ShaderStages can only take four elements before it starts to
+        // migrate. Make sure not to add redundant stages.
```

Comment:
If the problem is that we leak memory if the `ShaderStages` SmallVector ever grows, then wouldn't a better fix be to avoid a `SmallVector` (or any type that can move its memory) entirely? That is, should we adjust `PayloadAccessAnnotation` to store the stages in a fixed data structure instead?